src/main/java/hudson/plugins/gradle/AbstractBuildScanAction.java

```diff
@@ -40,11 +43,22 @@ public void addScanUrl(String scanUrl) {
         }
     }
 
+    public void addScanDetail(ScanDetail scanDetail) {
+        if (!scanDetails.contains(scanDetail)) {
```

Comment:
In order for this check to work we would need to override `equals` method of the `ScanDetail` class. Alternatively, I think we can use a `Map<String, ScanDetail>`, where the key is a scan URL. Then we can easily deduplicate by the URL.

Reply:
Could you also please add a simple unit test to ensure that we can't add duplicates?

Reply:
Well spotted 👍
I added a unit test.